test/utils/noderesourcetopologies/noderesourcetopologies.go

```diff
@@ -0,0 +1,228 @@
+		}
+		if initialQty.Cmp(updatedQty) != 0 {
+			klog.Infof("node %q zone %q resource %q initial=%v updated=%v", nodeName, zoneName, res.Name, initialQty, updatedQty)
+			return false, res.Name, nil
+		}
+	}
+	return true, "", nil
+}
+
+func checkConsumedResourcesAtLeast(resourcesInitial, resourcesUpdated []nrtv1alpha1.ResourceInfo, required corev1.ResourceList) (bool, error) {
+	for resName, resQty := range required {
+		initialQty, ok := findResourceAvailableByName(resourcesInitial, string(resName))
+		if !ok {
+			return false, fmt.Errorf("resource %q not found in the initial set", string(resName))
+		}
+		updatedQty, ok := findResourceAvailableByName(resourcesUpdated, string(resName))
+		if !ok {
+			return false, fmt.Errorf("resource %q not found in the updated set", string(resName))
+		}
+		expectedQty := initialQty.DeepCopy()
+		expectedQty.Sub(resQty)
+		ret := updatedQty.Cmp(expectedQty)
+		if ret > 0 {
+			return false, nil
+		}
+	}
+	return true, nil
+}
+
+func ResourcesFromPod(pod corev1.Pod) corev1.ResourceList {
+	res := make(corev1.ResourceList)
+	for idx := 0; idx < len(pod.Spec.Containers); idx++ {
+		cnt := &pod.Spec.Containers[idx] // shortcut
+		for resName, resQty := range cnt.Resources.Limits {
+			qty := res[resName]
+			qty.Add(resQty)
+			res[resName] = qty
+		}
+	}
+	return res
+}
+
+func AccumulateNRTNames(nrts []nrtv1alpha1.NodeResourceTopology) sets.String {
+	nodeNames := sets.NewString()
+	for _, nrt := range nrts {
+		nodeNames.Insert(nrt.Name)
+	}
+	return nodeNames
+}
+
+func FilterNRTWithTopologyManagerPolicy(nrts []nrtv1alpha1.NodeResourceTopology, tmPolicy nrtv1alpha1.TopologyManagerPolicy) []nrtv1alpha1.NodeResourceTopology {
+	ret := []nrtv1alpha1.NodeResourceTopology{}
+	for _, nrt := range nrts {
+		if !contains(nrt.TopologyPolicies, string(tmPolicy)) {
+			klog.Warningf("SKIP: node %q doesn't support topology manager policy %q", nrt.Name, string(tmPolicy))
+			continue
+		}
+		klog.Infof("ADD : node %q supports topology manager policy %q", nrt.Name, string(tmPolicy))
+		ret = append(ret, nrt)
+	}
+	return ret
+}
+
+func FilterNRTWithAnyZoneMatchingResources(nrts []nrtv1alpha1.NodeResourceTopology, requests corev1.ResourceList) []nrtv1alpha1.NodeResourceTopology {
+	ret := []nrtv1alpha1.NodeResourceTopology{}
+	for _, nrt := range nrts {
+		matches := 0
+		for _, zone := range nrt.Zones {
+			if !nodeResourceTopologyZoneResourcesMatchesRequest(zone.Resources, requests) {
+				continue
+			}
+			klog.Infof(" ----> node %q zone %q provides at least %#v", nrt.Name, zone.Name, requests)
+			matches++
+		}
+		if matches == 0 {
+			klog.Warningf("SKIP: node %q can't provide %#v", nrt.Name, requests)
+			continue
+		}
+		klog.Infof("ADD : node %q provides at least %#v", nrt.Name, requests)
+		ret = append(ret, nrt)
+	}
+	return ret
+}
+
+func FindNodeResourceTopologyFromList(nrts []nrtv1alpha1.NodeResourceTopology, name string) (*nrtv1alpha1.NodeResourceTopology, error) {
```

Comment:
The name of the package is already ```noderesourcetopologies``` so we can remove the name itself from the function.
Then it is going to be called like ```noderesourcetopologies.FindFromList```

Reply:
ack